libs/blocks/gnav/gnav-contextual-search.js

```diff
@@ -0,0 +1,87 @@
+import { fetchBlogArticleIndex } from '../article-feed/article-feed.js';
+import { getArticleTaxonomy, buildArticleCard } from '../article-feed/article-helpers.js';
+import { createTag } from '../../utils/utils.js';
+
+function highlightTextElements(terms, elements) {
+  elements.forEach((e) => {
+    const matches = [];
+    const txt = e.textContent;
+    terms.forEach((term) => {
+      const offset = txt.toLowerCase().indexOf(term);
+      if (offset >= 0) {
+        matches.push({ offset, term });
+      }
+    });
+    matches.sort((a, b) => a.offset - b.offset);
+    let markedUp = '';
+    if (!matches.length) markedUp = txt;
+    else {
+      markedUp = txt.substr(0, matches[0].offset);
+      matches.forEach((hit, i) => {
+        markedUp += `<mark class="gnav-search-highlight">${txt.substr(hit.offset, hit.term.length)}</mark>`;
+        if (matches.length - 1 === i) {
+          markedUp += txt.substr(hit.offset + hit.term.length);
+        } else {
+          markedUp += txt.substring(hit.offset + hit.term.length, matches[i + 1].offset);
+        }
+      });
```

Comment:
Nit: I often use `.reduce()` for looped text additions. It might clean things up here